[Do not merge] Fedora packaging

LecrisUT opened the pull request:
Now to handle the dependencies themselves an prepare for packaging

LecrisUT committed:
Prepare for packaging

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT committed:
Add remaining dependencies

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT commented:
I've started with `argon2` as a template on how to resolve the dependencies. Packit does not have proper dependencies support, but restarting the build should pick up the latest version

Also note the comments in commit 540aba24128c9a4fbcab448e52b28fa2951b28f2

LecrisUT commented:
Here is my workflow for adding these:
```console
$ export rust_pkg=...
$ mkdir ../rust-${rust_pkg} && cd ../rust-${rust_pkg} && touch rust2rpm.toml && rust2rpm ${rust_pkg}
```
After which I check https://bugzilla.redhat.com/ for any open review requests and https://crates.io/ for some other links.

- If the package has a review request, add a link, remove the `rust2rpm.toml`, and copy the spec file from the link
- If the package is already in fedora, copy the spec file and `rust2rpm.toml` (update the `.conf` file if necessary) and rerun `rust2rpm ${rust_pkg}`
- If none of the above are the case, proceed with the files generated

LecrisUT committed:
Reduce build to `fedora-rawhide`

Too many build dependencies to handle

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT commented:
/packit build

LecrisUT commented:
Downstream has made quite some progress, the main blocker was `sqlx` which has quite a lot of dependencies, and now there only `futures-intrusive` is blocking. The main dependencies we need to work on fixing are:
- [x] sqlx: https://github.com/LecrisUT/sqlx-rpmspec
  - [x] Submit for review
  - [x] In rawhide
- [x] rusty_paserk, rusty_paseto https://github.com/LecrisUT/rusty_parserk-paseto-rpmspec
  - [x] Submit for review
  - [x] In rawhide
- [x] metrics, metrics-exporter-prometheus https://github.com/LecrisUT/rust-metris-rpmspec
  - [x] Submit for review
  - [x] In rawhide
- [x] tonic https://github.com/LecrisUT/tonic-rpmspec
  - [x] Submit for review
  - [x] In rawhide
- [x] Other dependencies
  - [x] Submit for review
  - [ ] In rawhide 
- [x] atuin itself
  - [x] Submit for review
  - [ ] In rawhide

SriRamanujam commented:
> Downstream has made quite some progress, the main blocker was `sqlx` which has quite a lot of dependencies, and now there only `futures-intrusive` is blocking. The main dependencies we need to work on fixing are:
> 
>     * [x]  sqlx: https://github.com/LecrisUT/sqlx-rpmspec
> 
>     * [ ]  rusty_paserk
> 
>     * [ ]  rusty_paseto
> 
>     * [ ]  metrics
> 
>     * [ ]  metrics-exporter-prometheus
> 
>     * [ ]  interim
> 
>     * [ ]  cli-clipboard
> 
>     * [ ]  parse_duration: Probably need to patch dependency of `num`?

I'm also seeing that axum is failing to build. Looks like we may need to make some inquiries to downstream or patch the dependency down to 0.4.4. 

Thanks for your continued work on this!

LecrisUT commented:
Yeah, that whole dependency tree is being worked on [downstream](https://bugzilla.redhat.com/show_bug.cgi?id=2251680)

LecrisUT commented:
Status update:
- `sqlx` chain is basically finished
- `rusty_paserk`, `rusty_paseto` chain is almost finished
- `metrics` chain is almost finished
- `interim`, `cli-clipboard`, `parse_duration` have a simple dependency chain, should be straightforward to do (if my notes about it are still accurate)

LecrisUT committed:
Pull in other dependencies under review

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT commented:
Only remaining issue is with `cli-clipboard`: https://github.com/allie-wake-up/cli-clipboard/pull/21. We have an [alternative](https://github.com/atuinsh/atuin/pull/2067)

LecrisUT committed:
atuin-18.3.0 Fix dependencies

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT commented:
Seems the tests that are done now were not run in the vendored case because `cargo_install_lib 0`. But I don't see it referencing postgres in this case :/

SriRamanujam commented on .packit.yaml:
```diff
@@ -1,31 +1,121 @@
     upstream_package_name: atuin
-    downstream_package_name: rust-atuin
+    actions: &Actions-Package-rust2rpm-workspace
+      # Note: in upstream, the License has to be fixed manually after rust2rpm is run
```
For my own clarification: does "upstream" here refer to this repo, when generating the spec file for new patch releases?

LecrisUT replied:
I know I just wrote that sentence one week ago, but I forgot. But the gist of it is that the binary pulls in all other rust crates as static libraries so its license is the combination of all of them.

LecrisUT commented:
Notes to self: maybe the sqlite tests are fixed by: https://github.com/launchbadge/sqlx/issues/362#issuecomment-636661146

ellie commented:
Just a heads up that the actual fix for the SQLite test failure is to enable SQLITE_USE_URI in your SQLite build

Void had the same issue here: https://forum.atuin.sh/t/packaging-atuin-for-void-linux/20/9

LecrisUT commented:
Yes, I've opened a bug about it and asking other packagers for feedback: https://bugzilla.redhat.com/show_bug.cgi?id=2291157

LecrisUT commented:
@SriRamanujam btw, don't merge this one, because it would be good to have the vendored version for copr instead that is better supported.  Just need this on a separate branch or in the current PR as-is so I can link to the spec and srpm files. I plan to submit the rest of the package reviews the week after. This week is DevConf.

LecrisUT committed:
Add features.enable-all

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT committed:
Final review preparation

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT committed:
Fix bash-preexec

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT committed:
Review rust-tiny-bip39

LecrisUT committed:
Remove reviewed packages

LecrisUT committed:
Bump rust-minspan

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT committed:
Review rust-interim

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT committed:
Takeover rust-crypto_secretbox

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT committed:
Move builds to lecris's copr

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

LecrisUT committed:
Update `sqlx` dependency

Signed-off-by: Cristian Le <cristian.le@mpsd.mpg.de>

SriRamanujam commented:
CI passed!

LecrisUT commented:
Yeah, we're getting there, you can follow the progress [here](https://bugzilla.redhat.com/show_bug.cgi?id=2297310). I still need some experienced eyes to look over the `/etc/profile.d` addition because I think it needs to be gated against interactive environment and root user, similar to how Lmod is working. Or make the `/etc/profile.d` installation optional

LecrisUT committed:
Removed finished reviews

LecrisUT committed:
Update bash-preexec

LecrisUT committed:
Rerun `rust2rpm` on atuin

LecrisUT committed:
Separate the atuin package

LecrisUT commented:
I am splitting out the `bash-preexec` dependency in hopes to get this reviewed faster

LecrisUT committed:
Split the patches

LecrisUT committed:
Update tiny-bip39 dependency

LecrisUT committed:
Update tracing-tree dependency

LecrisUT committed:
Fix the build instruction

LecrisUT committed:
Address review comments

pinecat committed:
Add support for zsh

This patch adds the necesary lines in rust2rpm.toml and atuin.spec to
enable atuin support for users running the Z shell (zsh). If the
$ZSH_VERSION environment variable has been set (which is done by default
upon starting zsh), it will cause zsh to source the file
/usr/libexec/atuin/atuin-init.zsh, which contains the code normally
generated by 'atuin init zsh'.

The /etc/zprofile script calls 'emulate -L ksh', before sourcing files
in /etc/profile.d for compatibility reasons. However, atuin needs some
zsh-specific functions (mainly, compinit), so we call 'emulate -L zsh'
before running through the code in /usr/libexec/atuin/atuin-init.zsh,
and then call 'emulate -L ksh' at the very end.

pinecat committed:
Refactor how zsh's builtin 'emulate' command is used

This patch removes the second 'emulate [-L ksh]' call in favor of
passing the '-c' flag to run a single line of code outside of ksh.
Directly from the zshbuiltins(1) manpage: "If -c arg appears in flags,
arg is evaluated while the requested emulation is  temporarily  in
effect.  In this case the emulation mode and all options are restored to
their previous values before emulate returns."

The check for the zsh environment was also moved to an elif block,
instead of it's own if statement.

LecrisUT committed:
Merge pull request #4 from pinecat/review/rust-dependencies

Add support for zsh

LecrisUT commented:
Happy news: https://src.fedoraproject.org/rpms/atuin
